### PR TITLE
Update to apollo-compiler v1.0.0-beta.18

### PR DESCRIPTION
**Pull request by SimonSapin**
https://github.com/apollographql/apollo-rs/pull/868 introduced some breaking changes that cause a large but not complex diff here.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -48,7 +48,7 @@ debug = 1
 # Dependencies used in more than one place are specified here in order to keep versions in sync:
 # https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table
 [workspace.dependencies]
-apollo-compiler = "=1.0.0-beta.17"
+apollo-compiler = "=1.0.0-beta.18"
```
Comment (SimonSapin) [inline Cargo.toml:51]: This PR is a draft for now because this version does not exist yet

File: apollo-federation/src/link/argument.rs [hunk 2]
```diff
@@ -80,38 +79,6 @@ pub(crate) fn directive_required_string_argument(
     })
 }
 
-pub(crate) fn directive_optional_fieldset_argument(
```
Comment (SimonSapin) [inline apollo-federation/src/link/argument.rs:83]: This was the same as `directive_optional_string_argument`

File: apollo-federation/src/link/argument.rs [hunk 3]
```diff
@@ -80,38 +79,6 @@ pub(crate) fn directive_required_string_argument(
...
-            _ => Err(SingleFederationError::Internal {
-                message: format!("Invalid value for argument \"{}\": must be a string.", name),
-            }
-            .into()),
-        },
-        None => Ok(None),
-    }
-}
-
-pub(crate) fn directive_required_fieldset_argument(
```
Comment (SimonSapin) [inline apollo-federation/src/link/argument.rs:100]: This was the same as `directive_required_string_argument`

File: apollo-router/tests/integration/validation.rs [hunk 4]
```diff
@@ -128,7 +128,7 @@ async fn test_validation_error() {
           }
         },
         {
-          "message": "Field \"topProducts\" of type \"Product\" must have a selection of subfields. Did you mean \"topProducts { ... }\"?",
+          "message": "Field \"topProducts\" of type \"Query\" must have a selection of subfields. Did you mean \"topProducts { ... }\"?",
```
Comment (SimonSapin) [inline apollo-router/tests/integration/validation.rs:131]: The previous message was wrong. https://github.com/apollographql/apollo-rs/pull/865 fixed it.

**Review by goto-bus-stop — APPROVED** [review]
i at least skimmed every file and nothing really jumped out.

**Pull request merged**